### PR TITLE
Remove dataset pipeline from the Getting Started page

### LINKED ISSUE #23088: [Datasets] [Docs] Refactor docs structure + Getting Started vs. User Guide content
We are restructuring the Datasets docs into the following:
```
Ray Data
  - Getting Started with Ray Datasets
  - User Guides
    - Creating Datasets
    - Saving Datasets
    - Transforming Datasets
    - Exchanging Datasets
    - Pipelining Compute
    - ML Preprocessing
    - Working with Tensors
    - Performance Tips and Tuning
  - Key Concepts
  - Architecture
  - Example: Large-scale ML Ingest
  - Ray Datasets API
  - Integrations
    - Dask-on-Ray
    - Modin
    - Spark-on-Ray (RayDP)
    - Mars-on-Ray
```

## TODOs

- [x] Trim down Getting Started page to linear end-to-end example.
- [x] Remove DatasetPipelines from Getting Started page and de-emphasize/qualify throughout user guides (should only be used in advanced cases).
- [x] User Guide pages created/changed:
  - Creating Datasets - pulled from Getting Started
  - Saving Datasets - pulled from Getting Started
  - Transforming Datasets - pulled from Getting Started
  - Exchanging Datasets - pulled from Getting Started
  - Pipelining Compute - pulled from Getting Started
  - ML Preprocessing - rename "Data Loading and Preprocessing for ML Training"
- [x] Move third-party integrations to Integrations page
- [x] Update Ray Data landing page to contain the new User Guide page links (further updates to the landing page, e.g. overview of the data ecosystem, will happen in a future PR)

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
1. Dataset pipeline is advanced usage of Ray Dataset, which should not jam into the Getting Started page
2. We already have a separate/dedicated page called Pipelining Compute to cover the same content

## Related issue number
Closes #23088

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/getting-started.rst [hunk 1]
```diff
@@ -233,125 +233,3 @@ This is a common pattern useful for loading and splitting data between distribut
 
     ray.get([w.train.remote(s) for w, s in zip(workers, shards)])
     # -> [650, 650, ...]
-
-
-.. _dataset_pipelines_quick_start:
```
Comment (ericl) [inline doc/source/data/getting-started.rst:238]: Where is the content moving?

**Review by ericl — COMMENTED** [review]
Can you also add notes in the dataset pipeline page that you shouldn't use its shuffle unless you really need global shuffle, and add notes on alternatives with tf/torch datasets?

File: doc/source/data/getting-started.rst (hunk 1 above)
Comment (ericl) [reply doc/source/data/getting-started.rst:238]: (does it need to be added to front of datasets pipeline page)?

Comment (jianoaix) [reply doc/source/data/getting-started.rst:238]: Here: https://docs.ray.io/en/master/data/pipelining-compute.html#

**Comment by jianoaix** [comment]
> Can you also add notes in the dataset pipeline page that you shouldn't use its shuffle unless you really need global shuffle, and add notes on alternatives with tf/torch datasets?

Done.

**Review by ericl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**